target_chains/ethereum/entropy_sdk/solidity/IEntropyConsumer.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+abstract contract IEntropyConsumer {
+    // This address should be set to point at the Entropy contract. If not set, the
+    // contract will revert when it receives a callback from Entropy.
+    // Entropy address can be found here - https://docs.pyth.network/entropy/contract-addresses
+    address public entropy;
+
+    // This method is called by Entropy to provide the random number to the consumer.
```

Comment:
add comments to mention it asserts the msg.sender is entropy. Also mention that overrides should ALWAYS include this logic. Maybe even worth mentioning in the function name